src/Compiler/Checking/CheckPatterns.fs

```diff
@@ -35,6 +35,23 @@ type cenv = TcFileState
 // Helpers that should be elsewhere
 //-------------------------------------------------------------------------
 
+let inline findSynAttribute (attrName: string) (synAttrs: SynAttributes) =
```

Comment:
This is dangerous by not checking assembly of origin and also not respecting type aliases.
It should better be documented as such, to prevent usage in the compiler codebase.

(makes sense if you HAVE to work with untyped tree)


I also remember a new warning was added for AutoOpen if people were aliasing it.
If we are working with StructAttribute in an untyped way, then it should also better be guarded against aliasing... ?

Reply:
Let me check it, it's sealed, so it'll be good if we can forbid aliasing. I tried to make it work when we TC, but it were too much changes. I had to "abstract" option in compiler to be (Value)Option, but still TC/Unify correctly, but it was cascading everywhere.

Reply:
(documented here in the code for other programmers, to for users of F# compiler)

Reply:
Added a warning (extended current one). It will now warn on AutoOpen and on Struct aliasing.